### ISSUE TITLE
请问bs.npy是什么？

### ISSUE THREAD

**Issue by wning13** [ZiqiaoPeng/SyncTalk#18]

**Comment by lokvke** [comment]
as the paper said, bs means blendshape, you can check here:   https://github.com/psyai-net/EmoTalk_release

**Comment by wning13** [comment]
> as the paper said, bs means blendshape, you can check here: https://github.com/psyai-net/EmoTalk_release

谢谢，EmoTalk并没有提供获取Blendshape的方法，我选择了使用MediaPipe的方案来替代。

**Comment by johndpope** [comment]
I use mediapipe here - @wning13  - https://github.com/johndpope/Emote-hack/blob/main/HeadRotation.py
![image](https://github.com/ZiqiaoPeng/SyncTalk/assets/289994/7bf4c731-c596-4455-acee-59f876b0c656)

**Comment by ZiqiaoPeng** [comment]
Since the Blendshape Capturer of Emotalk has not been released, I am attempting to use some modules from Mediapipe as an alternative. I am verifying whether replacing the modules will lead to a decrease in performance.

**Comment by wning13** [comment]
Mediapipe 实现可以参考这个notebook https://colab.research.google.com/github/googlesamples/mediapipe/blob/main/examples/face_landmarker/python/%5BMediaPipe_Python_Tasks%5D_Face_Landmarker.ipynb

**Comment by Elysiameik** [comment]
> Mediapipe 实现可以参考这个notebook https://colab.research.google.com/github/googlesamples/mediapipe/blob/main/examples/face_landmarker/python/%5BMediaPipe_Python_Tasks%5D_Face_Landmarker.ipynb

blendshape是特定的52个值，如何从mediapipe的检测结果中变化成这特定的52个数值呢？

**Comment by Elysiameik** [comment]
> I use mediapipe here - @wning13 - https://github.com/johndpope/Emote-hack/blob/main/HeadRotation.py ![image](https://private-user-images.githubusercontent.com/289994/310418248-7bf4c731-c596-4455-acee-59f876b0c656.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3Nzc5MTksIm5iZiI6MTcwOTc3NzYxOSwicGF0aCI6Ii8yODk5OTQvMzEwNDE4MjQ4LTdiZjRjNzMxLWM1OTYtNDQ1NS1hY2VlLTU5Zjg3NmIwYzY1Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QwMjEzMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NmVjOWE3YzNlNzU5NmQwZGI1NGQyYTFkNDc1M2I3NDBiNDIxOTIwOTE3MTg4NTU2YzlmYjkwZjMxZWE4ZTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.do8KOngDV7lc3LstjVKv8FYFOOxVF0V7oD6x13QTCDA)

blendshape has a specific 52 values. How to change the detection results of mediapipe to these specific 52 values?

**Comment by jinqiupeter** [comment]
Is this the blendshape (from EmoTalk_release)?  https://github.com/psyai-net/EmoTalk_release/blob/main/demo.py#L55

**Comment by Elysiameik** [comment]
> Since the Blendshape Capturer of Emotalk has not been released, I am attempting to use some modules from Mediapipe as an alternative. I am verifying whether replacing the modules will lead to a decrease in performance.


I used the blendshapes generated by mediapipe, which  dimension is （6073,52）, same as the original bs.npy .I replaced the original bs.npy in May with bs.npy generated by mediapipe .The rest data in May remains. After training, May’s eyes did not blink.
Is using mediapipe the right solution?

**Comment by kike-0304** [comment]
> > > Since the Blendshape Capturer of Emotalk has not been released, I am attempting to use some modules from Mediapipe as an alternative. I am verifying whether replacing the modules will lead to a decrease in performance.
> > 
> > 
> > I used the blendshapes generated by mediapipe, which dimension is （6073,52）, same as the original bs.npy .I replaced the original bs.npy in May with bs.npy generated by mediapipe .The rest data in May remains. After training, May’s eyes did not blink. Is using mediapipe the right solution?
> 
> It seems there might be a problem with the loss function calculation for the eye. I'll make some modifications and test it Additionally, please note that the label orders between MediaPipe and EmoTalk are different.

Hello, have you successfully resolved this problem？

**Comment by kike-0304** [comment]
> 我在这里使用媒体管道 -@wning13 - https://github.com/johndpope/Emote-hack/blob/main/HeadRotation.py ![图像](https://private-user-images.githubusercontent.com/289994/310418248-7bf4c731-c596-4455-acee-59f876b0c656.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzOTg3NjQsIm5iZiI6MTcxMDM5ODQ2NCwicGF0aCI6Ii8yODk5OTQvMzEwNDE4MjQ4LTdiZjRjNzMxLWM1OTYtNDQ1NS1hY2VlLTU5Zjg3NmIwYzY1Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNFQwNjQxMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYTQyNWU1OTU5NDE3ZTNmMDcyYTU4ZGE0YjY4ZTQ1NDIyYWU3YzZlZGE5NzdlYjI1YTQ5MzVmYTI5OGQ0ZDA1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OWrJGCdCpZ9Na9CyC37RLbF33lKI5L_idqGcgsaZ4UQ)

Can you obtain the correct bs.npy file through this method?

**Comment by alexcazacu** [comment]
@wning13 did you manage to obtain good results using the blendshapes from MediaPipe?

**Comment by wning13** [comment]
> @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?

Yes, I've made three modifications:
1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
2. Utilizing `upface_mask` for the `loss_cross`.
3. Incrementing the lambda value of `loss_cross` suitably."

**Comment by lyq1152227095** [comment]
能分享一下具体的参数设计吗，我调整的参数但是一直无法眨眼

**Comment by Tiandishihua** [comment]
> > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> 
> Yes, I've made three modifications:
> 
> 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> 2. Utilizing `upface_mask` for the `loss_cross`.
> 3. Incrementing the lambda value of `loss_cross` suitably."

Can you share the processing code you get for bs.npy?

**Comment by wennjiee** [comment]
> > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> 
> Yes, I've made three modifications:
> 
> 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> 2. Utilizing `upface_mask` for the `loss_cross`.
> 3. Incrementing the lambda value of `loss_cross` suitably."

Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking.
the details I've done as follows, i wish you would share your modifacation

step3:  parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss")
step1:  parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye")
step2:  
        # eye att loss
        if self.opt.amb_eye_loss and not self.opt.torso:
            ambient_eye = outputs['ambient_eye']
            loss_cross = ((ambient_eye)*(upface_mask.view(-1))).mean()
            # loss_cross = ((ambient_eye)*(lowface_mask.view(-1))).mean()
            lambda_amb = step_factor * self.opt.lambda_amb
            loss += lambda_amb * loss_cross

**Comment by yunyu** [comment]
> > > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> > 
> > 
> > Yes, I've made three modifications:
> > 
> > 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> > 2. Utilizing `upface_mask` for the `loss_cross`.
> > 3. Incrementing the lambda value of `loss_cross` suitably."
> 
> Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking. the details I've done as follows, i wish you would share your modifacation
> 
> step3: parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss") step1: parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye") step2: # eye att loss if self.opt.amb_eye_loss and not self.opt.torso: ambient_eye = outputs['ambient_eye'] loss_cross = ((ambient_eye)_(upface_mask.view(-1))).mean() # loss_cross = ((ambient_eye)_(lowface_mask.view(-1))).mean() lambda_amb = step_factor * self.opt.lambda_amb loss += lambda_amb * loss_cross

Your code is increasing the general ambient loss lambda. You should just add a separate parameter to only adjust it for loss_cross

**Comment by wning13** [comment]
> > > > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> > > 
> > > 
> > > Yes, I've made three modifications:
> > > 
> > > 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> > > 2. Utilizing `upface_mask` for the `loss_cross`.
> > > 3. Incrementing the lambda value of `loss_cross` suitably."
> > 
> > 
> > Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking. the details I've done as follows, i wish you would share your modifacation
> > step3: parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss") step1: parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye") step2: # eye att loss if self.opt.amb_eye_loss and not self.opt.torso: ambient_eye = outputs['ambient_eye'] loss_cross = ((ambient_eye)_(upface_mask.view(-1))).mean() # loss_cross = ((ambient_eye)_(lowface_mask.view(-1))).mean() lambda_amb = step_factor * self.opt.lambda_amb loss += lambda_amb * loss_cross
> 
> Your code is increasing the general ambient loss lambda. You should just add a separate parameter to only adjust it for loss_cross

- The mask should be inverted.
- Keep the lambda of loss_amb_aud unchanged, while increasing the lambda of loss_cross by 10-100 times.

**Comment by yunyu** [comment]
> > > > > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> > > > 
> > > > 
> > > > Yes, I've made three modifications:
> > > > 
> > > > 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> > > > 2. Utilizing `upface_mask` for the `loss_cross`.
> > > > 3. Incrementing the lambda value of `loss_cross` suitably."
> > > 
> > > 
> > > Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking. the details I've done as follows, i wish you would share your modifacation
> > > step3: parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss") step1: parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye") step2: # eye att loss if self.opt.amb_eye_loss and not self.opt.torso: ambient_eye = outputs['ambient_eye'] loss_cross = ((ambient_eye)_(upface_mask.view(-1))).mean() # loss_cross = ((ambient_eye)_(lowface_mask.view(-1))).mean() lambda_amb = step_factor * self.opt.lambda_amb loss += lambda_amb * loss_cross
> > 
> > 
> > Your code is increasing the general ambient loss lambda. You should just add a separate parameter to only adjust it for loss_cross
> 
> * The mask should be inverted.
> * Keep the lambda of loss_amb_aud unchanged, while increasing the lambda of loss_cross by 10-100 times.

Can you clarify "the mask should be inverted"? I am training with upface_mask, is that it or should we do a 1-upface_mask? I am getting reasonable convergence although my models don't blink.

**Comment by wning13** [comment]
> > > > > > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> > > > > 
> > > > > 
> > > > > Yes, I've made three modifications:
> > > > > 
> > > > > 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> > > > > 2. Utilizing `upface_mask` for the `loss_cross`.
> > > > > 3. Incrementing the lambda value of `loss_cross` suitably."
> > > > 
> > > > 
> > > > Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking. the details I've done as follows, i wish you would share your modifacation
> > > > step3: parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss") step1: parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye") step2: # eye att loss if self.opt.amb_eye_loss and not self.opt.torso: ambient_eye = outputs['ambient_eye'] loss_cross = ((ambient_eye)_(upface_mask.view(-1))).mean() # loss_cross = ((ambient_eye)_(lowface_mask.view(-1))).mean() lambda_amb = step_factor * self.opt.lambda_amb loss += lambda_amb * loss_cross
> > > 
> > > 
> > > Your code is increasing the general ambient loss lambda. You should just add a separate parameter to only adjust it for loss_cross
> > 
> > 
> > 
> > * The mask should be inverted.
> > * Keep the lambda of loss_amb_aud unchanged, while increasing the lambda of loss_cross by 10-100 times.
> 
> Can you clarify "the mask should be inverted"? I am training with upface_mask, is that it or should we do a 1-upface_mask? I am getting reasonable convergence although my models don't blink.

``` python
if self.opt.amb_eye_loss and not self.opt.torso:
  ambient_eye = outputs['ambient_eye']
  loss_cross = ((ambient_eye)*(~upface_mask.view(-1))).mean()
  lambda_amb = step_factor * self.opt.lambda_amb_eye
  loss += lambda_amb * loss_cross
```

**Comment by yunyu** [comment]
> 
> ```python
> if self.opt.amb_eye_loss and not self.opt.torso:
>   ambient_eye = outputs['ambient_eye']
>   loss_cross = ((ambient_eye)*(~upface_mask.view(-1))).mean()
>   lambda_amb = step_factor * self.opt.lambda_amb_eye
>   loss += lambda_amb * loss_cross
> ```

Worked, thank you!
<img width="1435" alt="image" src="https://github.com/ZiqiaoPeng/SyncTalk/assets/8008350/e58411e0-34cd-4df4-947a-5819c57f85b5">

**Comment by wennjiee** [comment]
> > > > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> > > 
> > > 
> > > Yes, I've made three modifications:
> > > 
> > > 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> > > 2. Utilizing `upface_mask` for the `loss_cross`.
> > > 3. Incrementing the lambda value of `loss_cross` suitably."
> > 
> > 
> > Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking. the details I've done as follows, i wish you would share your modifacation
> > step3: parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss") step1: parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye") step2: # eye att loss if self.opt.amb_eye_loss and not self.opt.torso: ambient_eye = outputs['ambient_eye'] loss_cross = ((ambient_eye)_(upface_mask.view(-1))).mean() # loss_cross = ((ambient_eye)_(lowface_mask.view(-1))).mean() lambda_amb = step_factor * self.opt.lambda_amb loss += lambda_amb * loss_cross
> 
> Your code is increasing the general ambient loss lambda. You should just add a separate parameter to only adjust it for loss_cross

I don't understand "a spearate parameter for loss_cross", could you show more details about it. Or  I'd appreciate it if you could share your code corresponding to this part.

**Comment by wennjiee** [comment]
> > ```python
> > if self.opt.amb_eye_loss and not self.opt.torso:
> >   ambient_eye = outputs['ambient_eye']
> >   loss_cross = ((ambient_eye)*(~upface_mask.view(-1))).mean()
> >   lambda_amb = step_factor * self.opt.lambda_amb_eye
> >   loss += lambda_amb * loss_cross
> > ```
> 
> Worked, thank you! <img alt="image" width="1435" src="https://private-user-images.githubusercontent.com/8008350/326226853-e58411e0-34cd-4df4-947a-5819c57f85b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyODgxNTcsIm5iZiI6MTcxNDI4Nzg1NywicGF0aCI6Ii84MDA4MzUwLzMyNjIyNjg1My1lNTg0MTFlMC0zNGNkLTRkZjQtOTQ3YS01ODE5YzU3Zjg1YjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjhUMDcwNDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjZmZWJjN2VlZGY1MjFhNmMzNDM5MDA5ZWEzYTI2NjQ5MTg0MmU1YjhlNWRhNTlmYmFkYzE1MmI3ZTA1ZTIwMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zc8MWMnbOEpqXxuW8_KSBb949TGzsyR4T56NiFAyVQI">



> > > > @wning13 did you manage to obtain good results using the blendshapes from MediaPipe?
> > > 
> > > 
> > > Yes, I've made three modifications:
> > > 
> > > 1. Adjusting the `bs_area` parameter to `eye`. When set to `upper`, it necessitates a larger loss lambda, and more steps.
> > > 2. Utilizing `upface_mask` for the `loss_cross`.
> > > 3. Incrementing the lambda value of `loss_cross` suitably."
> > 
> > 
> > Hi, I use mediapipe to extract bs.npy. I chose the column corresponding to the eye during training. At the same time, I follow the steps 1, 2, 3 according to the method you said, but my loss is always around 0.0066 and it can't reduce from the start. It causes unstable head and no-eyeblinking. the details I've done as follows, i wish you would share your modifacation
> > step3: parser.add_argument('--lambda_amb', type=float, default=0.15, help="lambda for ambient loss") step1: parser.add_argument('--bs_area', type=str, default="eye", help="upper or eye") step2: # eye att loss if self.opt.amb_eye_loss and not self.opt.torso: ambient_eye = outputs['ambient_eye'] loss_cross = ((ambient_eye)_(upface_mask.view(-1))).mean() # loss_cross = ((ambient_eye)_(lowface_mask.view(-1))).mean() lambda_amb = step_factor * self.opt.lambda_amb loss += lambda_amb * loss_cross
> 
> Your code is increasing the general ambient loss lambda. You should just add a separate parameter to only adjust it for loss_cross

I modified my code with a separate parameter 'lambda_amb_eye' and I increase it by 10 - 100 times as mentioned above. And I changed the upface_mask. All modification as below, do you mean it or there's something i ignored. Thanks!
`    parser.add_argument('--lambda_amb_eye', type=float, default=5, help="lambda for ambient loss") # 1 - 10

      if self.opt.amb_eye_loss and not self.opt.torso:
          ambient_eye = outputs['ambient_eye']
          loss_cross = ((ambient_eye)*(~upface_mask.view(-1))).mean()
          # loss_cross = ((ambient_eye)*(lowface_mask.view(-1))).mean()
          lambda_amb = step_factor * self.opt.lambda_amb_eye
          loss += lambda_amb * loss_cross
`

**Comment by yunyu** [comment]
One more thing I noticed - it seems like the blendshape indexes don't quite match the emotalk ones. From Mediapipe, I get:

```
[(0, '_neutral'), (1, 'browDownLeft'), (2, 'browDownRight'), (3, 'browInnerUp'), (4, 'browOuterUpLeft'), (5, 'browOuterUpRight'), (6, 'cheekPuff'), (7, 'cheekSquintLeft'), (8, 'cheekSquintRight'), (9, 'eyeBlinkLeft'), (10, 'eyeBlinkRight'), (11, 'eyeLookDownLeft'), (12, 'eyeLookDownRight'), (13, 'eyeLookInLeft'), (14, 'eyeLookInRight'), (15, 'eyeLookOutLeft'), (16, 'eyeLookOutRight'), (17, 'eyeLookUpLeft'), (18, 'eyeLookUpRight'), (19, 'eyeSquintLeft'), (20, 'eyeSquintRight'), (21, 'eyeWideLeft'), (22, 'eyeWideRight'), (23, 'jawForward'), (24, 'jawLeft'), (25, 'jawOpen'), (26, 'jawRight'), (27, 'mouthClose'), (28, 'mouthDimpleLeft'), (29, 'mouthDimpleRight'), (30, 'mouthFrownLeft'), (31, 'mouthFrownRight'), (32, 'mouthFunnel'), (33, 'mouthLeft'), (34, 'mouthLowerDownLeft'), (35, 'mouthLowerDownRight'), (36, 'mouthPressLeft'), (37, 'mouthPressRight'), (38, 'mouthPucker'), (39, 'mouthRight'), (40, 'mouthRollLower'), (41, 'mouthRollUpper'), (42, 'mouthShrugLower'), (43, 'mouthShrugUpper'), (44, 'mouthSmileLeft'), (45, 'mouthSmileRight'), (46, 'mouthStretchLeft'), (47, 'mouthStretchRight'), (48, 'mouthUpperUpLeft'), (49, 'mouthUpperUpRight'), (50, 'noseSneerLeft'), (51, 'noseSneerRight')]
```

Right now we have in `provider.py`:
```
        if self.opt.bs_area == "upper":
            bs = np.hstack((bs[:, 0:5], bs[:, 8:10]))
        elif self.opt.bs_area == "single":
            bs = np.hstack((bs[:, 0].reshape(-1, 1),bs[:, 2].reshape(-1, 1),bs[:, 3].reshape(-1, 1), bs[:, 8].reshape(-1, 1)))
        elif self.opt.bs_area == "eye":
            bs = bs[:,8:10]
```

It seems like the indexes for eye should be replaced with 9:11 instead of 8:10.

**Comment by yunyu** [comment]
Ah yep, looks like that was done in the preprocessing code that was just released: https://github.com/ZiqiaoPeng/SyncTalk/commit/d89c6c77844b3655bb641a4842dd068616b4edaf#diff-a8a3c9a5ca89776de419860dbbf5a2ef712fb2d8219c4230b7f9b0fcf20fd4b1R66

**Comment by wennjiee** [comment]
> Ah yep, looks like that was done in the preprocessing code that was just released: [d89c6c7#diff-a8a3c9a5ca89776de419860dbbf5a2ef712fb2d8219c4230b7f9b0fcf20fd4b1R66](https://github.com/ZiqiaoPeng/SyncTalk/commit/d89c6c77844b3655bb641a4842dd068616b4edaf#diff-a8a3c9a5ca89776de419860dbbf5a2ef712fb2d8219c4230b7f9b0fcf20fd4b1R66)

Great! Thanks for anwsering. I also noticed the details about the bs.npy and replaced it with 9:11 for eyes encoding. Maybe there's something i ignored, I am checking my code for my reasonable loss.
